src/prover_disk.hpp

```diff
@@ -117,6 +119,33 @@ class DiskProver {
         delete[] c2_buf;
     }
 
+    explicit DiskProver(const std::vector<uint8_t>& vecBytes)
+    {
+        Deserializer deserializer(vecBytes);
+        deserializer >> version;
+        if (version != VERSION) {
+            // TODO: Migrate to new version if we change something related to the data structure
+            throw std::invalid_argument("DiskProver: Invalid version.");
+        }
+        deserializer >> filename;
+        deserializer >> memo;
+        deserializer >> id;
+        deserializer >> k;
+        deserializer >> table_begin_pointers;
+        deserializer >> C2;
+    }
+
+    DiskProver(DiskProver&& other) noexcept
+    {
+        std::lock_guard<std::mutex> lock(other._mtx);
```

Comment:
> It seems unlikely that any other thread could (legitimately) have a reference to this object, since it's an r-value reference.

Good point, see 068feffcc0802c8f563c2f58f9979a7010917501.